cpmpy/transformations/linearize.py

```diff
@@ -100,11 +108,23 @@ def linearize_constraint(lst_of_expr, supported={"sum","wsum"}, reified=False):
                 # BV -> LinExpr
                 elif isinstance(cond, _BoolVarImpl):
                     lin_sub = linearize_constraint([sub_expr], supported=supported, reified=True)
-                    newlist += [cond.implies(lin) for lin in lin_sub]
+                    for lin in lin_sub:
+                        # linearize might return True/False for trivial constraints
+                        if is_true_cst(lin):
```

Comment:
Can you add a TODO here to check performance of this check? The happy-path is now the longest one which might make the transformation slower.